### DISCUSSION TITLE
I can't seem to figure out how to use the --filter option

### DISCUSSION THREAD

**Discussion by saladuit** [Snaipe/Criterion#477] [Q&A] [answered]
I've tried numerous things and eventually I've given up. So I'm turning here to perhaps get some guidance.
This is the name of my test suite and test cases, but for some reason the filter option doesn't let me filter them.
```
TestSuite(expand_dollar);

Test(expand_dollar, basic)
{
// test stuff
}
```
I've tried the following commands:
 ./test.out -f [dollar_expand::basic:]
./test.out -f dollar_expand::basic: 
./test.out -f dollar_expand
 ./test.out -f  
./test.out -f slkdfj/* 
./test.out -f expand_tokens::multiple_expansions:
./test.out -f expand_dollar::basic: 
 ./test.out expand_dollar::basic:  
 ./test.out --filter expand_dollar 
 ./test.out --filter expand_dollar::  
etc etc etc a whole bunch more
I thought it went somewhere when I tried -f instead of --filter, but it doesn't seem to work. 
Help is welcome!
edit: now I at least see that -f is that the first test fails

**Comment by saladuit** [comment]
![image](https://user-images.githubusercontent.com/43225170/221314035-1e35c34a-eff1-477b-a642-02facc6aec60.png)

**Comment by saladuit** [comment]
![image](https://user-images.githubusercontent.com/43225170/221314178-7599fd1a-7375-458c-9207-caa9ddb0ad6f.png)

**Comment by saladuit** [comment]
![image](https://user-images.githubusercontent.com/43225170/221315317-d18c3cc2-a07b-4142-a3e9-d46e0bdf4002.png)

**Comment by MrAnno** [comment, marked as answer]
In case of `Test(expand_dollar, basic)`, the filter expression you’re looking for is `expand_dollar/basic`. You can use wildcards as well: `test.out --filter '*basic'`.

**Comment by saladuit** [comment]
@Snaipe I really suggest adding explicitly that quotation marks are needed!

**Reply by MrAnno** [reply]
Your shell expands wildcards to files/directories, so quotes are needed. We can add a few examples to the documentation.

**Reply by saladuit** [reply]
That would be great! I'm not sure why it didn't work then. I really thought I had also tried this. But oh well.
